### PR TITLE
fix build logs loading, alignment and pending state messages

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/HAC-2808
https://issues.redhat.com/browse/HAC-3025
https://issues.redhat.com/browse/HAC-3071

## Description

UI doesn't tell user when the selected log isn't available any more. Due to the pod limit, the pruner will start to remove the old pods in order to create new pods, so the user will not be able to view the logs for the pods that are pruned already.

This PR fixes few issues around the build logs:

1. Display the message in the UI to notify that the logs are no longer available for the <selected_task>.
2. Display the pending state message if the task is not started yet.
3. Clicking on the build log should show the last completed task logs.

Note: Pods limit are now increased to 150 (130 for taskruns , 20 for deployments), so the pruner will kick in when the pods count reaches 130

## Type of change

- [x] Bugfix

## Screen shots / Gifs for design review 

Clone repository task's pod was pruned in the below screenshot:

Screenshot 2023-02-03 at 4 31 40 PM

Task's Pending state message: 

Screenshot 2023-02-03 at 2 42 27 PM

Fixed Alignment of the task list and log window:

alignment-after

## Unit tests

```
  PipelineRunLogs
    ✓ should render task list and log window (58 ms)
    ✓ should render select and render the first task log (18 ms)
    ✓ should render waiting for logs message (7 ms)
    ✓ should no logs found message when pipelinerun status is misisng (4 ms)
    ✓ should render logs no longer available message when pods are not available (17 ms)
 ```
 
 ```
   getRenderContainers
    ✓ should render default values if invalid value is passed (2 ms)
    ✓ should render containers for the given pod (1 ms)
 
 ```

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc: @beaumorley @christianvogt

### CODE REVIEW

File: src/shared/components/pipeline-run-logs/PipelineRunLogs.tsx [hunk 1]
```diff
@@ -98,6 +99,11 @@ class PipelineRunLogs extends React.Component<PipelineRunLogsProps, PipelineRunL
         namespace: obj.metadata.namespace,
         isList: false,
       };
+
+    const waitingForPods = !!(activeItem && !resource);
+    const taskName = get(taskRunFromYaml, [activeItem, 'pipelineTaskName'], '-');
+    const pipelineRunFinished = ![runStatus.Running].includes(pipelineRunStatus);
```
Comment (jeff-phillips-18) [inline src/shared/components/pipeline-run-logs/PipelineRunLogs.tsx:105]: Why check against inclusion in an array?

File: src/shared/components/pipeline-run-logs/logs/LogsWrapperComponent.tsx [hunk 2]
```diff
@@ -18,12 +18,21 @@ const LogsWrapperComponent: React.FC<LogsWrapperComponentProps> = ({ resource, .
...
+  } else if (error) {
+    resourceRef.current = null;
   }
 
-  return resourceRef.current ? (
-    <MultiStreamLogs {...props} resourceName={resource.name} resource={resourceRef.current} />
-  ) : (
-    <LoadingBox />
+  let errorMessage;
+  if ((error as HttpError)?.code === 404) {
```
Comment (jeff-phillips-18) [inline src/shared/components/pipeline-run-logs/logs/LogsWrapperComponent.tsx:26]:
```suggestion
  const errorMessage = error as HttpError)?.code === 404) ? 
```
Prefer ternary operator

File: src/shared/components/pipeline-run-logs/PipelineRunLogs.tsx (hunk 1 above)
Comment (karthikjeeyar) [reply src/shared/components/pipeline-run-logs/PipelineRunLogs.tsx:105]: Initially had multiple statuses to check, but later realised its not required and missed to refactor this. Updated now

**Review by jeff-phillips-18 — APPROVED** [review]

**Pull request merged**